### ISSUE TITLE
Test accuracy changes with test batch size

### ISSUE THREAD

**Issue by shanyucha** [TimoSaemann/caffe-segnet-cudnn5#22]
### Issue summary
validation accuracy changes when batch_size value changes.
opened in bvlc caffe. https://github.com/BVLC/caffe/issues/5621#issuecomment-368446563
### Steps to reproduce
see above link.
I'm using caffe-segnet (the newest version). While do validation during training, I found that the test accuracy changes when I change batch size in TEST phase.
